src/types.ts

```diff
@@ -33,33 +38,19 @@ export type RequireOnlyOne<T, Keys extends keyof T = keyof T> = Pick<T, Exclude<
 
 /* Unknown Record */
 export type UR = Record<string, unknown>;
-export type UnknownType = UR; //alias to avoid breaking change
-
-export type DefaultGenerics = {
-  attachmentType: UR;
-  channelType: UR;
-  commandType: LiteralStringForUnion;
-  eventType: UR;
-  memberType: UR;
-  messageType: UR;
-  pollOptionType: UR;
-  pollType: UR;
-  reactionType: UR;
-  userType: UR;
-};
-
-export type ExtendableGenerics = {
-  attachmentType: UR;
-  channelType: UR;
-  commandType: string;
-  eventType: UR;
-  memberType: UR;
-  messageType: UR;
-  pollOptionType: UR;
-  pollType: UR;
-  reactionType: UR;
-  userType: UR;
-};
+export type UnknownType = UR; // alias to avoid breaking change
+
+export interface CustomAttachmentType {}
+export interface CustomChannelType {}
+export interface CustomCommandType {}
```

Comment:
Might have to figure out a better way for this to work. Currently using `keyof CustomCommandType` to make it work with interface merging.

Reply:
This right here: https://github.com/GetStream/stream-chat-js/pull/1458/files#diff-c54113cf61ec99691748a3890bfbeb00e10efb3f0a76f03a0fd9ec49072e410aR2124

Actual module augmentation would have to look something like:
```ts
// file.d.ts

import 'stream-chat';

declare module 'stream-chat' {
    interface CustomCommandType {
        customCommand: unknown;
        otherCustomCommand: unknown;
    }
}
```

Reply:
Looks good ! We're already doing similar things for other parts of the RN SDK (extending functionality and overloads) so this will come natural.

The entire change is breaking I assume, right ? What's the strategy to go about merging this when the time comes ? I suppose all SDKs would need to bump their major versions if they wish to upgrade their version of `stream-chat`. Although kind of large in size, the change seems easy to do (if generics were being used in the first place).

Reply:
Yes, in case of React SDK we'd be releasing v13.0.0 which will bump its `stream-chat` peer to v9.0.0 - the peer of the SDK is currently bound to v8-something.

As for the migration - see this draft of a [migration guide here](https://github.com/GetStream/docs-content/pull/167), feel free to suggest enhancements. Not sure where it'll live yet, I assume both React and RN will be similar in terms of migration but I'm open to adding tabs to switch between RN and React.

Reply:
That's fair - in that case we should pick it up earlier rather than later since `stream-chat` is a direct dependency for `stream-chat-react-native` (and not a peer one).

Since we'll likely have to do a V7 to go along with it as well, maybe we can do the same ? it should get rid of our pesky TS errors when there's a version mismatch (despite the fact it's a direct dependency).

fyi @oliverlaz

Reply:
@isekovanic I agree! 

We should have a similar guide, and also we need to update the following chapters:
- https://getstream.io/chat/docs/sdk/react/guides/typescript_and_generics/
- https://getstream.io/chat/docs/sdk/react-native/customization/typescript/